test/plots/scatterPlotTests.ts

```diff
@@ -2,395 +2,399 @@
+        plot.renderTo(svg);
+
+        let entities = plot.entitiesAt({ x: xScale.scale(1), y: yScale.scale(1) });
+        assert.lengthOf(entities, 1, "only one Entity has been retrieved");
+        assert.deepEqual(entities[0].datum, { x: 1, y: 1 }, "correct datum has been retrieved");
+
+        svg.remove();
+      });
+
+      it("determines whether an Entity contains a given Point by its position and size", () => {
+        plot.size(10);
+        plot.renderTo(svg);
+
+        let entities = plot.entitiesAt({ x: xScale.scale(1) + 5, y: yScale.scale(1) - 5});
+        assert.lengthOf(entities, 1, "only one Entity has been retrieved");
+        assert.deepEqual(entities[0].datum, { x: 1, y: 1 }, "correct datum has been retrieved");
+
+        plot.size(6);
+        entities = plot.entitiesAt({ x: xScale.scale(1) + 5, y: yScale.scale(1) - 5});
+        assert.lengthOf(entities, 0, "none of the Entities is retrieved");
+        svg.remove();
+      });
+
+      it("returns all Entities containing a given Point across all Datasets", () => {
+        plot.addDataset(new Plottable.Dataset([{ x: 0, y: 1 }, { x: 200, y: 200 }]));
+        plot.renderTo(svg);
+
+        let entities = plot.entitiesAt({ x: xScale.scale(0.5), y: yScale.scale(0.5) });
+        assert.lengthOf(entities, 3, "all 3 Entities containing the Point have been retrieved");
+        assert.deepEqual(entities[0].datum, { x: 0, y: 0 }, "correct datum has been retrieved");
+        assert.deepEqual(entities[1].datum, { x: 1, y: 1 }, "correct datum has been retrieved");
+        assert.deepEqual(entities[2].datum, { x: 0, y: 1 }, "correct datum has been retrieved");
+
+        svg.remove();
+      });
+    });
+
+    describe("Invalid data and deprecated method", () => {
+      let svg: d3.Selection<void>;
+      let plot: Plottable.Plots.Scatter<number, number>;
+      beforeEach(() => {
+        svg = TestMethods.generateSVG();
+        plot = new Plottable.Plots.Scatter<number, number>();
+        plot.x((d: any) => d.x)
+            .y((d: any) => d.y);
+      });
+
+      it.skip("correctly handles NaN, undefined, Infinity, and non-number x and y values", () => {
+        let data = [
+          { x: 0.0, y: 0.0 },
+          { x: 0.2, y: 0.2 },
+          { x: 0.4, y: 0.4 },
+          { x: 0.6, y: 0.6 },
+          { x: 0.8, y: 0.8 }
+        ];
+        let dataset = new Plottable.Dataset(data);
+        plot.addDataset(dataset);
+        plot.renderTo(svg);
+
+        assert.strictEqual(plot.selections().size(), 5, "all 5 data points are drawn");
+
+        let dataWithError = data.slice();
+        dataWithError[2] = { x: 0.4, y: NaN };
+        dataset.data(dataWithError);
+        assert.strictEqual(plot.selections().size(), 4, "only 4 data points are drawn when one of the y values is NaN");
+        assert.lengthOf(plot.entities(), 4, "only 4 Entities returned  when one of the y values is NaN");
+
+        dataWithError = data.slice();
```

Comment:
Instead of creating so many objects, can we keep amending one item at a time and see the `selections().size()` go down all the time.

Don't have a strong preference here though.